apps/OpenEnergyMonitor/myheatpump/myheatpump.js

```diff
@@ -639,7 +654,11 @@ function powergraph_load()
             } else {
                 data["heatpump_elec"] = feed.getdata(feeds["heatpump_elec"].id,view.start,view.end,view.interval,1,0,skipmissing,limitinterval);
             }
-            powergraph_series.push({label:"Electric", data:data["heatpump_elec"], yaxis:1, color:1, lines:{show:true, fill:0.3, lineWidth:0.5}});
+            if (all_same_interval) {
+                powergraph_series.push({label:"Electric", data:remove_null_values(data["heatpump_elec"]), yaxis:1, color:1, lines:{show:true, fill:0.3, lineWidth:0.5}});
+            } else {
+                powergraph_series.push({label:"Electric", data:data["heatpump_elec"], yaxis:1, color:1, lines:{show:true, fill:0.3, lineWidth:0.5}});
+            }
```

Comment:
This fixes a bug whereby the electric graph would sometimes disappear when plotting Carnot or COP.